applications/dashboard/models/class.invitationmodel.php

```diff
@@ -79,21 +80,27 @@ public function Save($FormPostValues, $UserModel, $Options = array()) {
          $InviteCount = $UserModel->GetInvitationCount($UserID);
          if ($InviteCount == 0) {
             $this->Validation->AddValidationResult('Email', 'You do not have enough invitations left.');
-            return FALSE;
+            return false;
          }
 
          // Make sure that the email does not already belong to an account in the application.
          $TestData = $UserModel->GetWhere(array('Email' => $Email));
          if ($TestData->NumRows() > 0) {
             $this->Validation->AddValidationResult('Email', 'The email you have entered is already related to an existing account.');
-            return FALSE;
+            return false;
          }
 
          // Make sure that the email does not already belong to an invitation in the application.
          $TestData = $this->GetWhere(array('Email' => $Email));
          if ($TestData->NumRows() > 0) {
-            $this->Validation->AddValidationResult('Email', 'An invitation has already been sent to the email you entered.');
-            return FALSE;
+            if (!$Resend) {
+               $this->Validation->AddValidationResult('Email', 'An invitation has already been sent to the email you entered.');
+               return false;
+            } else {
+               // Remove the old invitation.
+               $DeleteID = val('InvitationID', $TestData->FirstRow(DATASET_TYPE_ARRAY));
+               $this->Delete($DeleteID);
```

Comment:
If there is another error then the invitation will not be saved, but the old invitation will still be deleted. You need to only delete if there are no errors.